test/e2e/kcp_remediations.go

```diff
@@ -0,0 +1,693 @@
+
+	return allMachines.List(), newMachines.List()
+}
+
+// getServerAddr returns the address to be used for accessing the management cluster from a workload cluster.
+func getServerAddr(kubeconfigPath string) string {
+	kubeConfig, err := clientcmd.LoadFromFile(kubeconfigPath)
+	Expect(err).ToNot(HaveOccurred(), "failed to load management cluster's kubeconfig file")
+
+	clusterName := kubeConfig.Contexts[kubeConfig.CurrentContext].Cluster
+	Expect(clusterName).ToNot(BeEmpty(), "failed to identify current cluster name in management cluster's kubeconfig file")
+
+	serverAddr := kubeConfig.Clusters[clusterName].Server
+	Expect(serverAddr).ToNot(BeEmpty(), "failed to identify current server address in management cluster's kubeconfig file")
+
+	// On CAPD, if not running on Linux, we need to use Docker's proxy to connect back to the host
+	// to the CAPD cluster. Moby on Linux doesn't use the host.docker.internal DNS name.
+	if runtime.GOOS != "linux" {
+		serverAddr = strings.ReplaceAll(serverAddr, "127.0.0.1", "host.docker.internal")
+	}
+	return serverAddr
+}
+
+// getAuthenticationToken returns a bearer authenticationToken with minimal RBAC permissions to access the mhc-test ConfigMap that will be used
+// to control machines bootstrap during the remediation tests.
+func getAuthenticationToken(ctx context.Context, managementClusterProxy framework.ClusterProxy, namespace string) []byte {
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mhc-test",
+			Namespace: namespace,
+		},
+	}
+	Expect(managementClusterProxy.GetClient().Create(ctx, sa)).To(Succeed(), "failed to create mhc-test service account")
+
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mhc-test",
+			Namespace: namespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"get", "list", "patch"},
+				APIGroups:     []string{""},
+				Resources:     []string{"configmaps"},
+				ResourceNames: []string{"mhc-test"},
+			},
+		},
+	}
+	Expect(managementClusterProxy.GetClient().Create(ctx, role)).To(Succeed(), "failed to create mhc-test role")
+
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mhc-test",
+			Namespace: namespace,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				APIGroup:  "",
+				Name:      "mhc-test",
+				Namespace: namespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.SchemeGroupVersion.Group,
+			Kind:     "Role",
+			Name:     "mhc-test",
+		},
+	}
+	Expect(managementClusterProxy.GetClient().Create(ctx, roleBinding)).To(Succeed(), "failed to create mhc-test role binding")
+
+	cmd := exec.CommandContext(ctx, "kubectl", fmt.Sprintf("--kubeconfig=%s", managementClusterProxy.GetKubeconfigPath()), fmt.Sprintf("--namespace=%s", namespace), "create", "token", "mhc-test") //nolint:gosec
```

Comment:
I think we should be able to do this either with client-go or the CR SubResource client `managementClusterProxy.GetClient().SubResource("token").Create(ctx, ...)`

Could be a follow-up cleanup issue if it's more effort to get this to work

Reply:
It is more complex, it requires creating a secret and some more work, but no objection to doing it